ListApp/ListApp.Api.Tests/ItemsController.cs

```diff
@@ -0,0 +1,349 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using System.Threading.Tasks;
+using System.Web.Http;
+using System.Web.Http.Results;
+using JsonPatch;
+using NUnit.Framework;
+using ListApp.Api.Controllers.V1;
+using ListApp.Api.Models;
+using ListApp.Api.Utils;
+
+namespace ListApp.Api.Tests
+{
+    [TestFixture]
+    public class ItemsControllerTests
+    {
+        private static readonly Guid TheGuid = Guid.Parse("00000000-0000-0000-0000-000000000003");
+        private const string PostedItemText = "Build a monument";
+        private static readonly ListItem PostedItem = new ListItem { Id = TheGuid, Text = PostedItemText };
+        private static readonly Func<Guid> GuidCreator = () => TheGuid;
+
+        private ItemsController _theController;
+
+        [SetUp]
+        public void SetUp()
+        {
+            ItemsController.InitializeItems();
+            _theController = new ItemsController(GuidCreator);
+        }
+
+        // Some of the tests require to create an instance of Http server/client,
+        // as passing direct invalid method argument is not possible (GUID) OR 
+        // since the validity check is done by action filters, not the controller.
+
+        #region GET tests
+        
+        [Test]
+        public async Task Get_ReturnsAllDefaultItems()
+        {
+            var receivedItems = await _theController.GetItems();
+            Assert.That(receivedItems, Is.EqualTo(Utils.Constants.MockListItems).Using(new ListItemEqualityComparer()));
+        }
+
+        // It is not possible to send a invalid GUID object, but it can easily happen when using the API
+        // The server/client has to be used to test this.
+        // The argument validity check is handled by the action filter
+        [Test]
+        public async Task Get_Returns400OnInvalidGuidFormat()
+        {
+            var config = new HttpConfiguration();
+            config = ServerInit.InitializeConfiguration(config);
+
+            using (var server = new HttpServer(config))
+            using (var client = new HttpClient(server))
+            {
+                var receivedResponse = await client.GetAsync("http://localhost:57187/api/v1/items/1");
```

Comment:
I don't think you should test the server...it doesn't really look like a unit test.